File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    answer = False
    stk = []
    for i in s:
        if i == '(':
            stk.append(1)
        else:
            try:
                stk.pop()
            except:
                return answer
    if not stk:
        answer = True
    return answer
    

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.java
import java.util.Stack;

class Solution {
    boolean solution(String s) {
        Stack<Character> st = new Stack<>();
        for (int i = 0; i < s.length(); i++) {
            if (s.charAt(i) == '(') {
                st.push(s.charAt(i));
            }
            else {
                if (!st.empty() && st.peek() == '(') {
                    st.pop();
                }
                else {
                    return false;
                }
            }
        }

        return st.empty() ? true : false;
    }
}